src/printer.js

```diff
@@ -418,8 +418,9 @@ function genericPrintNoParens(path, options, print, args) {
 
       const shouldInline =
         (firstNonMemberParent &&
-          ((firstNonMemberParent.type === "VariableDeclarator" &&
-            firstNonMemberParent.id.type !== "Identifier") ||
+          (firstNonMemberParent.type === "NewExpression" ||
```

Comment:
What about CallExpression?

Reply:
@azz I'm not sure what you meant.. what about CallExpression?

Reply:
As in the same without `new`. It actually already behaves as expected, but I'm not sure why.

```js
function functionName() {
  // indent to make the line break
  if (true) {
    this._aVeryLongVariableNameToForrrrrrrBreak = this.Promise(
      (resolve, reject) => {
        // do something
      }
    );
  }
}
```

Reply:
Ah, understood now! That's because that case doesn't get here, actually. That's printed by `printMemberChain` (see https://github.com/prettier/prettier/blob/master/src/printer.js#L889-L893)

Reply:
Ah, of course!